src/App.jsx

```diff
@@ -3,21 +3,41 @@ import reactLogo from './assets/react.svg'
 import viteLogo from '/vite.svg'
 import './App.css'
 import './TodoList'
+import { useEffect } from 'react'
 import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 import Search from './Search'
+
+// Custom hook to persist state in localStorage
+const useSemiPersistentState = () => {
+  // Initialize 'todoList' with the value from localStorage or an empty array if not found
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || [])
+  
+  // Update localStorage every time 'todoList' changes
+  useEffect(() => {
+    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
```

Comment:
If localStorage.getItem('savedTodoList') returns null, JSON.parse(null) will throw an error. You can handle this by providing a default value directly in JSON.parse:
```
const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList') || '[]'));
```

Reply:
I fixed the error